doc/changelog.md

```diff
@@ -9,30 +9,50 @@
   [(#650)](https://github.com/PennyLaneAI/catalyst/pull/650)
   [(#649)](https://github.com/PennyLaneAI/catalyst/pull/649)
   [(#661)](https://github.com/PennyLaneAI/catalyst/pull/661)
+  [(#621)](https://github.com/PennyLaneAI/catalyst/pull/621)
 
   Catalyst now supports callbacks with parameters and return values.
   The following is now possible:
 
   ```py
-  @pure_callback
+  @debug.callback
   def foo(val):
     return val
 
+  @pure_callback
+  def bar(val) -> int:
+    return val + 1
+
   @qjit
   def circuit(param):
-    return foo(param)
+    x = bar(param)
+    foo(x)
 
   ```
 
   ```pycon
   >>> print(circuit(123))
-  123
+  124
   ```
 
   This includes support for the specialized `pure_callback` and `debug.callback` where
   `pure_callback` is expected to return a value and be side effect free,
   while `debug.callback` is expected to produce a side effect and have no return values.
 
+  Syntactic sugar on top of `debug.callback` includes `debug.print` which allows the user
+  to use f-stringlike to format the string before printing.
```

Comment:
I would compare it `str.format()` rather than fstrings because its behaviour is closer to that right? 
`print("{c} {b} {a}", a=a, b=b, c=c)` ~ `print("{c} {b} {a}".format(a=a, b=b, c=c))`

Reply:
Might even make sense to link to the doc page: https://docs.python.org/3/library/string.html#formatstrings

Reply:
```suggestion
  to use formatstrings to format the string before printing. See https://docs.python.org/3/library/string.html#formatstrings
```